components/tools/ToolsDashboard.tsx

```diff
@@ -149,18 +136,30 @@ export default function ToolsDashboard() {
       }
     });
 
-    setToolsList(tempToolsList);
-  };
+    return tempToolsList;
+  }, [isPaid, isAsyncAPIOwner, languages, technologies, categories, searchName]);
+
+  // useEffect to scroll to the opened category when url has category as element id
+  useEffect(() => {
+    const { hash } = window.location;
 
+    if (hash) {
+      const elementID = decodeURIComponent(hash.slice(1));
+      const element = document.getElementById(elementID);
+
+      if (element) {
+        document.documentElement.style.scrollPaddingTop = '6rem';
```

Comment:
Instead of directly modifying the DOM, can't we use inbuilt hooks of react to interact with virtual DOM? 

Modifying real DOM in react is a bad practice.

Reply:
Agree! We are now using react refs to focus the category divs. 
but for the scroll padding we need the body element, I will have to pass the ref of the of the body element to ToolDashboard page. Should I go ahead and do it as well?